pkg/rpcclient/wsclient.go

```diff
@@ -712,6 +712,11 @@ func (c *WSClient) makeWsRequest(r *neorpc.Request) (*neorpc.Response, error) {
 func (c *WSClient) performSubscription(params []any, rcvr notificationReceiver) (string, error) {
 	var resp string
 
+	if flt := rcvr.Filter(); flt != nil {
+		if err := flt.IsValid(); err != nil {
+			return "", err
+		}
+	}
```

Comment:
And the client-side functionality should also be tested. There's already test for VM states, need to add a test for notification name. See `TestWSExecutionVMStateCheck`.